api-documentation/shacl.md

```diff
@@ -0,0 +1,181 @@
+# Using SHACL to describe data structures
+
+## Story
+
+As an API publisher
+
+I want to use SHACL to describe data structures
+
+So that consumers can use existing tooling to consume the API
+
+## Details
+
+SHACL, which stands for Shapes Constraint Language is an RDF vocabulary, defined in a [W3C recommendation][shacl], which is used to describe graph shapes.
+
+It is very expressive, more rich than `hydra:Class` et.al., and out of the box provides multiple features to provide an accurate description of request payloads.
+
+[Many uses for SHACL have been proposed][ucr] such as, but not limited to, validation and building user interfaces dynamically.
+
+This page shows how Shapes can be used to replace or extend Hydra Core vocabulary terms within the API Documentation and resource representations.
+
+[shacl]: https://www.w3.org/TR/shacl/
+[ucr]: https://www.w3.org/TR/shacl-ucr/
+
+### Announce SHACL in API Documentation
+
+An API which chooses to use SHACL in addition to the built-in `hydra:Class` `MUST` announce that fact in its API Documentation resource
+
+```json
+{
+  "@context": {
+    "hydra": "http://www.w3.org/ns/hydra/core#",
+    "requires": { "@id": "hydra:requires", "@type": "@id" }
+  },
+  "@id": "/api",
+  "@type": "hydra:ApiDocumentation",
+  "requires": "http://www.w3.org/ns/hydra/shacl"
```

Comment:
No strong opinion on the `hydra:requires` term and its semantics. 

Alternatively I thought `hydra:uses` or `hydra:imports`

Reply:
I think better approach would be just to point to the SHACL's namespace, `http://www.w3.org/ns/shacl#`

Reply:
Maybe 🤔

As seen below, my intention was for `http://www.w3.org/ns/hydra/shacl` to mean "SHACL extensions" and itself would become a namespace for additional terms

Reply:
Ah, I remember now. I was thinking that maybe the requirements could be more fine-grained like

`<> hydra:requires 
  hashi:OperationExpects, 
  hashi:IriTemplate`

But coming back to my first answer still. The meaning is that "the API requires SHACL extension". Which is slightly more specific than just SHACL

Reply:
I'm not convinced it's good direction. I think hydra should be extended in places where it does not provide any tools - replacing them is not a good move. It's better to create a separate spec.

Reply:
> I think hydra should be extended in places where it does not provide any tools - replacing them is not a good move

What do you mean by that? I do propose a separate spec, auxiliary to core

```diff
-http://www.w3.org/ns/hydra/core#
+http://www.w3.org/ns/hydra/shacl#
```

Reply:
I don't like an idea terms built in hydra would be replaced with some custom ones. In your example, `hashi:IriTemplate` would probably replace `hydra:IriTemplate`. I think it's not a approach

Reply:
Oh, no, I do not propose to replace anything. 

I only wanted to propose a more  fine-grained choice of which parts of the API are described with SHACL. `hashi:IriTemplate` indeed may have been an unfortunate choice

Reply:
>`hashi:IriTemplate` indeed may have been an unfortunate choice

Indeed - unfortunate. Could we come with a better one?

Reply:
Ugh, is that really necessary? It's not even mentioned in the pull request.

For the time being let's assume that only HASHI as a whole can be required, imported or used. However we choose.